### PR TITLE
refactor: Use the latest helm chart create cluster

### PR DESCRIPTION
**Pull request by daviderli614**

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -1,15 +1 @@
 package utils
-
```
Comment (zyy17) [inline pkg/utils/utils.go:2]: If the `utils.go` is not used anymore, delete it. Don't leave any unused files.

File: pkg/index/index.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
+package index
```
Comment (zyy17) [inline pkg/index/index.go]: the package `index` is not a good name.

File: pkg/index/index.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/blang/semver/v4"
+	"gopkg.in/yaml.v2"
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+const (
+	defaultIndexURL = "https://raw.githubusercontent.com/GreptimeTeam/helm-charts/gh-pages/index.yaml"
+)
+
+type ChartVersions []*ChartVersion
```
Comment (zyy17) [inline pkg/index/index.go]: Can we just import `ChartVersion` and `IndexFile` from the Helm library? If can't, use `third_party/` to import external code.

File: pkg/index/index.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	"sort"
+	"time"
+
+	"github.com/blang/semver/v4"
+	"gopkg.in/yaml.v2"
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+const (
+	defaultIndexURL = "https://raw.githubusercontent.com/GreptimeTeam/helm-charts/gh-pages/index.yaml"
```
Comment (zyy17) [inline pkg/index/index.go]: I think it should be `GreptimeHelmChartsIndex`. Because it's not configurable.

File: pkg/index/index.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+func GetUrlFromRemoteIndex(chartName string) (string, error) {
+	rsp, err := http.Get(defaultIndexURL)
+	if err != nil {
+		return "", err
+	}
+	defer rsp.Body.Close()
+
+	body, _ := ioutil.ReadAll(rsp.Body)
```
Comment (zyy17) [inline pkg/index/index.go]: need to handle error

File: pkg/index/index.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	// TillerVersionDeprecated is deprecated in Helm 3, and therefore ignored. However, with a strict
+	// YAML parser enabled, this field must be present.
+	TillerVersionDeprecated string `yaml:"tillerVersion,omitempty"`
+
+	// URLDeprecated is deprecated in Helm 3, superseded by URLs. It is ignored. However,
+	// with a strict YAML parser enabled, this must be present on the struct.
+	URLDeprecated string `yaml:"url,omitempty"`
+}
+
+func GetUrlFromRemoteIndex(chartName string) (string, error) {
```
Comment (zyy17) [inline pkg/index/index.go]: It's not a good implementation, what you want to do is:

- fetch metadata from the index file;
- get the latest version from metadata;
- get the download URL from the metadata of the latest version;

It's better to design the struct and implement the key methods of the struct above.

File: pkg/manager/manager.go [hunk 7]
```diff
@@ -64,9 +58,8 @@ type DeleteClusterOption struct {
 }
 
 type CreateOperatorOptions struct {
-	Namespace       string
```
Comment (zyy17) [inline pkg/manager/manager.go:67]: It's not the correct implementation. If the user specifies the version, use the specific version; if not, use the latest version. DON'T delete the version field.

**Comment by zyy17** [comment]
Can you add the unit tests for your code?

**Comment by daviderli614** [comment]
@zyy17 I will define a test function to call GetUrlFromRemoteIndex

File: pkg/index/index.go (hunk 2 above)
Comment (daviderli614) [reply pkg/index/index.go]: I will write GetLatestChart in helm.go

**Pull request closed without merging**